astropy/modeling/bounding_box.py

```diff
@@ -9,13 +9,24 @@
 import abc
 import copy
 import warnings
-from typing import Any, Callable, NamedTuple
+from typing import TYPE_CHECKING, NamedTuple
 
 import numpy as np
 
 from astropy.units import Quantity
 from astropy.utils import isiterable
 
+if TYPE_CHECKING:
+    from typing import Any, Callable
+
+    from astropy.units import UnitBase
+    from astropy.utils.compat import PYTHON_LT_3_11
+
+    if PYTHON_LT_3_11:
+        from typing_extensions import Self
+    else:
+        from typing import Self
```

Comment:
IMO in a TYPE_CHECKING setting we can assume `typing_extensions` is installed. No runtime breakage if it's not.
`typing_extensions` has a python3.11 check internally, obviating the need here, and `ruff`'s pyupgrade rules will auto-fix `typing_extensions` -> `typing` when our min-version is 3.11.

Reply:
Agreed.